### ISSUE TITLE
extract frames from mp4 pipeline

### ISSUE THREAD

**Issue by Joe-Surrey** [open-mmlab/mmpose#459]
I wanted to extract frames from an mp4 file so I wrote this
Sharing in case anyone finds it useful (or knows a better way)

    @PIPELINES.register_module()
    class LoadImageFromMP4:
         """Loading MP4 from file.
	    image_file in json must be in the format mp4file%frame
            e.g. car_video.mp4%6 for frame 6 of car_video.mp4
        """
    
        def __init__(self,):
            self.readers = {}
    
    
        def __call__(self, results):
            """Loading image from file."""
            video_file,frame = results['image_file'].split("%")
            if video_file not in self.readers:
                self.readers[video_file] = mmcv.VideoReader(video_file)
            
            img = self.readers[video_file].get_frame(int(frame))
    
            if img is None:
                raise ValueError('Fail to read frame {} of {}'.format(frame,image_file))
    
            results['img'] = img
            return results

**Comment by innerlee** [comment]
If the dataset is very large, retaining so many reader objects might use out the OS's number of opened files quota, right?

**Comment by Joe-Surrey** [comment]
yeah maybe it would be good to delete the reader after all of its frames have been read
